Src/Support/GoogleApis.Tests/Apis/Download/MediaDownloaderTest.cs

```diff
@@ -39,95 +39,154 @@ namespace Google.Apis.Tests.Apis.Download
     [TestFixture]
     class MediaDownloaderTest
     {
-        /// <summary>The content the "server" uses to send to the client.</summary>
+        /// <summary>A content string that will be returned and looked for.</summary>
         private static readonly byte[] MediaContent = Encoding.UTF8.GetBytes("Media content goes here. This is an example of test content.");
 
-        /// <summary> An handler which demonstrate a client download which contains multiple chunks. </summary>
-        private class MultipleChunksMessageHandler : CountableMessageHandler
+        /// <summary>
+        /// An error object that will be returned and looked for.
+        /// </summary>
+        private static readonly RequestError BadRequestError = new RequestError
         {
-            /// <summary>Gets or sets the chunk size to send.</summary>
-            public int ChunkSize { get; set; }
+            Code = 12345,
+            Message = "Bad Request!",
+            Errors = new[] { new SingleError { Message = "The request was malformed." } }
+        };
 
-            /// <summary>Gets or sets the status code in case we simulate an error.</summary>
-            public HttpStatusCode StatusCode { get; set; }
+        /// <summary>
+        /// An error string that will be returned and looked for.
+        /// </summary>
+        private const string NotFoundError = "No resource found by that name.";
 
-            /// <summary>Gets or sets the cancellation token used to cancel a request in the middle.</summary>
-            public CancellationTokenSource CancellationTokenSource { get; set; }
+        /// <summary>
+        /// Get an available port. This is best-effort since the port may immediately be
+        /// used by another process.
+        /// </summary>
+        /// <returns>A port number</returns>
+        private static int GetOpenPort()
+        {
+            var tcpListener = new TcpListener(IPAddress.Loopback, 0);
+            try
+            {
+                tcpListener.Start();
+                return (tcpListener.LocalEndpoint as IPEndPoint).Port;

```

Comment:
Tiny nit: I'd prefer a cast here, so that if it's ever *not* an IPEndPoint, we'll get a InvalidCastException instead of NullReferenceException. Just one of my bugbears :)